src/content/app/genome-browser/components/browser-cog/BrowserCogList.tsx

```diff
@@ -14,42 +14,55 @@
  * limitations under the License.
  */
 
-import React, { useEffect, memo } from 'react';
-import { useDispatch, useSelector } from 'react-redux';
+import React, { useEffect, memo, useRef } from 'react';
+
 import {
-  UpdateTrackSummaryAction,
   IncomingActionType,
-  TrackSummaryList,
-  TrackSummary
+  type UpdateTrackSummaryAction,
+  type TrackSummaryList,
+  type TrackSummary
 } from '@ensembl/ensembl-genome-browser';
 
 import useGenomeBrowser from 'src/content/app/genome-browser/hooks/useGenomeBrowser';
 
 import BrowserCog from './BrowserCog';
 
+import { useAppDispatch, useAppSelector } from 'src/store';
 import {
   getBrowserCogList,
-  getBrowserCogTrackList,
   getBrowserSelectedCog
 } from 'src/content/app/genome-browser/state/track-config/trackConfigSelectors';
 import {
-  CogList,
-  updateCogTrackList,
+  type CogList,
+  updateCogList,
   updateSelectedCog
 } from 'src/content/app/genome-browser/state/track-config/trackConfigSlice';
+import {
+  getBrowserActiveFocusObjectId,
+  getBrowserActiveGenomeId
+} from 'src/content/app/genome-browser/state/browser-general/browserGeneralSelectors';
 
 import styles from './BrowserCogList.scss';
 
 export const BrowserCogList = () => {
-  const browserCogList = useSelector(getBrowserCogList);
-  const browserCogTrackList = useSelector(getBrowserCogTrackList);
-  const selectedCog = useSelector(getBrowserSelectedCog);
-
-  const dispatch = useDispatch();
+  const browserCogList = useAppSelector(getBrowserCogList);
+  const selectedCog = useAppSelector(getBrowserSelectedCog);
+  const genomeId = useAppSelector(getBrowserActiveGenomeId);
+  const objectId = useAppSelector(getBrowserActiveFocusObjectId);
+  const dispatch = useAppDispatch();
 
+  const genomeIdRef = useRef(genomeId);
   const { genomeBrowser } = useGenomeBrowser();
 
+  useEffect(() => {
+    genomeIdRef.current = genomeId;
+  }, [genomeId]);
```

Comment:
What‘s the purpose of assigning genome id to a ref?

If you need access to the genome id as it existed during the previous render cycle, then we have a `usePrevious` hook. But looking at the code, I don't even understand the intention behind this change.

Reply:
Removed the ref.